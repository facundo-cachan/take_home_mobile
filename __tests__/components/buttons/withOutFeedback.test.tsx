/**
 * @format
 */
import { ButtonWithoutFeedback } from '@components';
import { describe, expect } from '@jest/globals';
import {
  fireEvent,
  screen
} from '@testing-library/react-native';
import { toJSON, withThemeProvider } from '__tests__/utils/customRender';

const onPressMock = jest.fn();
const props = {
  color: 'red',
  style: { backgroundColor: 'blue' },
  text: 'Button Test',
  icon: 'home',
  iconSize: 30,
  onPress: onPressMock,
};
const Btn = () => withThemeProvider(<ButtonWithoutFeedback {...props} />)

describe('ButtonWithoutFeedback', () => {
  beforeEach(() => {
    // Mock fetch with a resolved Promise that returns sample data
    // @ts-ignore
    jest.clearAllMocks();
    // @ts-ignore
    global.fetch = jest.fn().mockResolvedValue({
      json: () => Promise.resolve([])
    });
    toJSON(<Btn />)
  });
  test('Snapshot', () => {
    expect(screen).toMatchSnapshot()
  })
  test('Renders correctly', () => {
    const icon = screen.getByText(props.text)
    expect(icon).toBeTruthy()
    expect(icon.props.style).toBe(props.style)
  })
  test('Calls onPress when pressed', () => {
    const icon = screen.getByText(props.text)
    fireEvent(icon, 'press')
    expect(onPressMock).toHaveBeenCalled()
  })
})