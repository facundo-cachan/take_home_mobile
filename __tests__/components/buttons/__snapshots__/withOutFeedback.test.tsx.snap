// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithoutFeedback Snapshot 1`] = `
<View
  accessibilityLabel="home"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "display": "flex",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "red",
          "fontSize": 30,
        },
        {
          "backgroundColor": "blue",
        },
        {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    ï€•
  </Text>
  <Text
    style={
      {
        "backgroundColor": "blue",
      }
    }
  >
    Button Test
  </Text>
</View>
`;
